pkg/server/authorizedentries/entries.go

```diff
@@ -0,0 +1,52 @@
+package authorizedentries
+
+import (
+	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"google.golang.org/protobuf/proto"
+)
+
+type entryRecord struct {
+	EntryID  string
+	ParentID string
+	SPIFFEID string
+
+	// Pointer to the entry. For cloning only after the end of the crawl.
+	EntryCloneOnly *types.Entry
+}
+
+func entryRecordByEntryID(a, b entryRecord) bool {
+	switch {
+	case a.EntryID < b.EntryID:
+		return true
+	case a.EntryID > b.EntryID:
+		return false
+	default:
+		return a.ParentID < b.ParentID
```

Comment:
maybe I'm missing something, but this means that we have 2 entries with the same EntryID? so technically we may never get into this right?

Reply:
No, you are right. I'll clean this up.